alacritty_terminal/src/grid/storage.rs

```diff
@@ -51,81 +52,40 @@ pub struct Storage<T> {
 
 impl<T: PartialEq> PartialEq for Storage<T> {
     fn eq(&self, other: &Self) -> bool {
-        // Make sure length is equal
-        if self.inner.len() != other.inner.len() {
-            return false;
-        }
-
-        // Check which vec has the bigger zero
-        let (ref bigger, ref smaller) =
-            if self.zero >= other.zero { (self, other) } else { (other, self) };
+        // Both storage buffers need to be truncated and zeroed
+        assert_eq!(self.zero, 0);
+        assert_eq!(other.zero, 0);
 
-        // Calculate the actual zero offset
-        let bigger_zero = bigger.zero;
-        let smaller_zero = smaller.zero;
-
-        // Compare the slices in chunks
-        // Chunks:
-        //   - Bigger zero to the end
-        //   - Remaining lines in smaller zero vec
-        //   - Beginning of smaller zero vec
-        //
-        // Example:
-        //   Bigger Zero (6):
-        //     4  5  6  | 7  8  9  | 0  1  2  3
-        //     C2 C2 C2 | C3 C3 C3 | C1 C1 C1 C1
-        //   Smaller Zero (3):
-        //     7  8  9  | 0  1  2  3  | 4  5  6
-        //     C3 C3 C3 | C1 C1 C1 C1 | C2 C2 C2
-        let len = self.inner.len();
-        bigger.inner[bigger_zero..]
-            == smaller.inner[smaller_zero..smaller_zero + (len - bigger_zero)]
-            && bigger.inner[..bigger_zero - smaller_zero]
-                == smaller.inner[smaller_zero + (len - bigger_zero)..]
-            && bigger.inner[bigger_zero - smaller_zero..bigger_zero]
-                == smaller.inner[..smaller_zero]
+        self.inner == other.inner && self.len == other.len
```

Comment:
I don't really like this implementation tbh, since it doesn't make sense to me, that something should be truncated before `eq`. However since `eq` is only used by our tests, maybe it's not a bad idea to keep things simple here, but in this case we should state it.

Reply:
It's just unnecessary complexity if we never use it without truncation anyways.

We're also not really implementing a full equality check properly, but when it makes sense to compare them, this is the most reasonable way.

Reply:
I mean we should state that this thing is only for tests, right?